src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3885,16 +3885,30 @@ private int GetOrCreatePlaceholderSlot(object identifier, TypeWithAnnotations ty
                 return null;
             }
 
+            bool isInferred = node.Syntax.Kind() == SyntaxKind.ImplicitArrayCreationExpression;
+            var arrayType = VisitArrayInitialization(node.Type, isInferred, initialization, node.HasErrors);
+            SetResultType(node, TypeWithState.Create(arrayType, NullableFlowState.NotNull));
+            return null;
+        }
+
+        private TypeSymbol VisitArrayInitialization(TypeSymbol type, bool isInferred, BoundArrayInitialization initialization, bool hasErrors)
```

Comment:
>VisitArrayInitialization [](http://example.com/codeflow?start=27&length=24)

The name feels somewhat misleading because it looks like the helper is used for types other than arrays.

Reply:
The bound node is called `BoundArrayInitialization`, so this is following our normal naming convention